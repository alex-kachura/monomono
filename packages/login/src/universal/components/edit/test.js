import React from 'react';
import { fromJS } from 'immutable';
import { EditPage } from './';

describe('[Component: EditPage]', () => {
  let component;
  let mockFieldChange;
  let mockFieldBlur;
  let mockSaveData;
  let mockGetErrorMessage;
  const mockErrorMsg = 'mock-error-msg';

  let mockProps = {
    fields: fromJS([
      {
        id: 'sample1',
        name: 'sample1',
        label: 'pages.edit.fields.sample1.label',
        isValid: true,
        value: '',
        type: 'input',
        hasBlurred: false,
        constraints: [
          {
            type: 'mandatory',
            text: 'pages.edit.fields.sample1.errors.empty',
            validator: true,
            isValid: true,
          },
        ],
      },
    ]),
    csrf: 'mock-csrf',
    getLocalePhrase: (key) => key,
  };

  beforeEach(() => {
    mockFieldChange = jest.fn();
    mockFieldBlur = jest.fn();
    mockSaveData = jest.fn();
    mockGetErrorMessage = jest.fn(() => mockErrorMsg);

    mockProps = {
      ...mockProps,
      saveData: mockSaveData,
      fieldChange: mockFieldChange,
      fieldBlur: mockFieldBlur,
    };

    jest.doMock('../../../utils/validation', () => ({
      getErrorMessage: mockGetErrorMessage,
    }));

    component = global.contextualShallow(<EditPage {...mockProps} />);
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('submitting the form', () => {
    it('should call saveData correctly', () => {
      component.find('form').simulate('submit', { preventDefault: () => false });

      expect(mockSaveData).toHaveBeenCalledWith('sampleData');
    });
  });

  describe('blurring field', () => {
    it('should trigger fieldBlur methd', () => {
      component.find('#sample1-input').simulate('blur', {
        target: {
          value: 'mockValue1',
        },
      });

      expect(mockFieldBlur).toHaveBeenCalledWith({
        fieldName: 'sample1',
        value: 'mockValue1',
        hasBlurred: true,
      });
    });
  });

  describe('changing field', () => {
    it('should trigger fieldChange methd', () => {
      component.find('#sample1-input').simulate('change', {
        target: {
          value: 'mockValue2',
        },
      });

      expect(mockFieldChange).toHaveBeenCalledWith({
        fieldName: 'sample1',
        value: 'mockValue2',
        hasBlurred: false,
      });
    });
  });
});
