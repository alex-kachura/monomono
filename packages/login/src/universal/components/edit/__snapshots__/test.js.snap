// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: EditPage] should render correctly 1`] = `
<SideEffect(DocumentTitle)
  title="pages.landing.title"
>
  <div>
    <PageTitle
      margin={true}
      tagName="h1"
    >
      pages.edit.title
    </PageTitle>
    <styled.div>
      <form
        autoComplete="off"
        method="POST"
        noValidate={true}
        onSubmit={[Function]}
      >
        <Styled(FormGroup)
          error={false}
          errorMessage=""
          id="sample1"
          key="pages.edit.fields.sample1.label"
          labelText="pages.edit.fields.sample1.label"
          required={true}
        >
          <Input
            disabled={false}
            domRef={[Function]}
            error={false}
            id="sample1-input"
            name="sample1"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value=""
          />
        </Styled(FormGroup)>
        <input
          name="_csrf"
          type="hidden"
          value="mock-csrf"
        />
        <Styled(WithProps(Button))
          type="submit"
        >
          pages.edit.submit-btn
        </Styled(WithProps(Button))>
      </form>
    </styled.div>
  </div>
</SideEffect(DocumentTitle)>
`;
